### PR TITLE
Document aeios.AES.rekey() and refactor arg validation

### PR DESCRIPTION
**Pull request by isacben**
Hi,

This PR solves: aeios.AES.rekey() is not documented #7435 

a) Documented the aeios.AES.rekey() function
b) Refactored the arguments validation. Please let me know if that was the idea of "canonicalize arg checking".

Tested with `adafruit_feather_rp2040` with the following code:

```python
import aesio
import os
from binascii import hexlify, unhexlify

plaintext = unhexlify("6bc1bee22e409f96e93d7e117393172a")
key = unhexlify("2b7e151628aed2a6abf7158809cf4f3c")
iv = unhexlify("000102030405060708090a0b0c0d0e0f")

cyphertext = bytearray(len(plaintext))
output = memoryview(cyphertext)
cipher = aesio.AES(key=key, mode=aesio.MODE_CBC, IV=iv)

# Encrypt
cipher.encrypt_into(plaintext, output)

print(end='\n\n')
print('input:     ', str(hexlify(plaintext)), end='\n')
print('encrypted: ', str(hexlify(output)), end='\n')

# Decrypt
plaintext = bytearray(len(plaintext))
cipher = aesio.AES(key=key, mode=aesio.MODE_CBC, IV=iv)

output = memoryview(plaintext)
cipher.decrypt_into(cyphertext, output)
print('decrypted: ', str(hexlify(output)), end='\n')

# Rekey
plaintext = unhexlify("6bc1bee22e409f96e93d7e117393172a")
key2 = unhexlify("2b7e151628aed2a6abf7158809cAAAAA")
cyphertext = bytearray(len(plaintext))
output = memoryview(cyphertext)

cipher.rekey(key=key2, IV=iv) # update key of the existing cipher object

# Encrypt
cipher.encrypt_into(plaintext, output)

print(end='\n')
print('Rekey...', end='\n')
print('input:     ', str(hexlify(plaintext)), end='\n')
print('encrypted: ', str(hexlify(output)), end='\n')

# Decrypt
plaintext = bytearray(len(plaintext))
cipher = aesio.AES(key=key2, mode=aesio.MODE_CBC, IV=iv)

output = memoryview(plaintext)
cipher.decrypt_into(cyphertext, output)
print('decrypted: ', str(hexlify(output)), end='\n\n')
```

Output:

```
input:      b'6bc1bee22e409f96e93d7e117393172a'
encrypted:  b'7649abac8119b246cee98e9b12e9197d'
decrypted:  b'6bc1bee22e409f96e93d7e117393172a'

Rekey...
input:      b'6bc1bee22e409f96e93d7e117393172a'
encrypted:  b'ff7e2905e0af08d89121429ee15b311b'
decrypted:  b'6bc1bee22e409f96e93d7e117393172a'
```

Thank you!

### CODE REVIEW

File: shared-bindings/aesio/aes.c [hunk 1]
```diff
@@ -98,33 +98,49 @@ STATIC mp_obj_t aesio_aes_make_new(const mp_obj_type_t *type, size_t n_args,
     return MP_OBJ_FROM_PTR(self);
 }
 
-STATIC mp_obj_t aesio_aes_rekey(size_t n_args, const mp_obj_t *pos_args) {
+//|     def rekey(
+//|         self,
+//|         key: ReadableBuffer,
+//|         iv: Optional[ReadableBuffer] = None,
```
Comment (tannewt) [inline shared-bindings/aesio/aes.c:104]:
```suggestion
//|         IV: Optional[ReadableBuffer] = None,
```

Comment (tannewt) [reply shared-bindings/aesio/aes.c:104]: What capitalization is correct here? It looks like CP will parse `IV`.

**Review by tannewt — CHANGES_REQUESTED** [review]
One question. Looks good otherwise. Thank you!

File: shared-bindings/aesio/aes.c (hunk 1 above)
Comment (isacben) [reply shared-bindings/aesio/aes.c:104]: You are right. Will fix that. Thank you!

**Review by tannewt — CHANGES_REQUESTED** [review]
One more spot. Thanks!

File: shared-bindings/aesio/aes.c [hunk 2]
```diff
@@ -98,33 +98,49 @@ STATIC mp_obj_t aesio_aes_make_new(const mp_obj_type_t *type, size_t n_args,
...
-STATIC mp_obj_t aesio_aes_rekey(size_t n_args, const mp_obj_t *pos_args) {
+//|     def rekey(
+//|         self,
+//|         key: ReadableBuffer,
+//|         IV: Optional[ReadableBuffer] = None,
+//|     ) -> None:
+//|         """Update the AES state with the given key.
+//|
+//|         :param ~circuitpython_typing.ReadableBuffer key: A 16-, 24-, or 32-byte key
+//|         :param ~circuitpython_typing.ReadableBuffer iv: Initialization vector to use
```
[the remark this answers is not in the document]
Comment (isacben) [reply shared-bindings/aesio/aes.c:109]: Oops, missed that! Thanks for pointing it out!

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**